src/taipy/core/notification/notifier.py

```diff
@@ -0,0 +1,43 @@
+# Copyright 2023 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+from queue import Queue
+from typing import List, Tuple
+
+from .event import Event
+from .registration import Registration
+from .topic import Topic
+
+
+class Notifier:
+    _registrations: List[Registration] = []
+
+    @classmethod
+    def register(cls, entity_type, entity_id, operation, attribute_name) -> Tuple[str, Queue]:
+        register_id = "gui"  # TODO generate an id to return so the client can unregister.
```

Comment:
Since the user will be using this register_id, it should be simple and straight-forward. I'm thinking of something like all the parameters combined perhaps f"{entity_type}_{entity_id}_{operation}_{attribute_name}", but this is quite long :/ . We can also ask the user to provide a name that they prefer?

Reply:
we will generate a unique id on our side first. We will allow users to provide their own id later on but not in this ticket. (User-provided id must be unique)